apps/remote_control/lib/lexical/remote_control/code_mod/rename.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Lexical.RemoteControl.CodeMod.Rename do
+  alias Lexical.Ast.Analysis
+  alias Lexical.Document.Edit
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.RemoteControl.CodeMod.Rename.Prepare
+
+  @spec prepare(Analysis.t(), Position.t()) ::
+          {:ok, {atom(), String.t()}, Range.t()} | {:error, term()}
+  def prepare(%Analysis{} = analysis, %Position{} = position) do
+    Prepare.prepare(analysis, position)
+  end
+
+  @renamable_mapping %{module: __MODULE__.Module}
+
+  @spec rename(Analysis.t(), Position.t(), String.t()) ::
+          {:ok, %{Lexical.uri() => [Edit.t()]}} | {:error, term()}
+  def rename(%Analysis{} = analysis, %Position{} = position, new_name) do
+    case Prepare.resolve(analysis, position) do
+      {:ok, {renamable, entity}, range} ->
+        rename_module = @renamable_mapping[renamable]
```

Comment:
> shouldn't you see if the thing can be renamed before you prepare it? 

`Prepare.resolve` does exactly that. When it returns `{:ok, tuple, range}`, it means that this is an entity we support for renaming.

Therefore, I think using `@renaming_mapping[renamable]` here is fine, as it will certainly succeed unless we wrote the keys in `renaming_mapping` incorrectly, but that would be easily discovered by testing.